benches/benches/reachability.rs

```diff
@@ -223,7 +229,10 @@ fn benchmark_hydroflow(c: &mut Criterion) {
                             .flatten()
                             .copied();
 
-                        let pull = origins.chain(possible_reach).filter(|v| seen.insert(*v));
+                        let mut seen_state = context.state_ref(seen_handle).unwrap().borrow_mut();
```

Comment:
Is using a `RefCell` like this the preferred pattern? If yes, why do we have `state_mut` at all?

Reply:
So it is possible to use `state_mut`... however you have to be careful since then that prevents you from getting state from two handles simultaneously, since they might actually be the same handle.